api/v1beta1/glanceapi_types.go

```diff
@@ -107,6 +107,34 @@ type GlanceAPISpec struct {
 	// +kubebuilder:validation:Optional
 	// ExtraMounts containing conf files and credentials
 	ExtraMounts []GlanceExtraVolMounts `json:"extraMounts"`
+
+	// +kubebuilder:validation:Optional
+	// NetworkAttachments list of network attachment definitions the service pod gets attached to
+	NetworkAttachments []string `json:"networkAttachments"`
+
+	// +kubebuilder:validation:Optional
+	// ExternalEndpoints, expose a VIP via MetalLB on the pre-created address pool
+	ExternalEndpoints []MetalLBConfig `json:"externalEndpoints"`
+}
+
+// MetalLBConfig to configure the MetalLB loadbalancer service
+type MetalLBConfig struct {
```

Comment:
@stuggi quick question: `ExternalEndpoints` is optional in the CR, which is ok, but as long as it's defined, I'm wondering if we should make sure that the `Endpoint` (`internal` or `public`) is always defined (otherwise it should fail). 
Should then we need to apply the same pattern at least to the `IP` address ?
I'm basically ok about keeping the struct optional, but when it's defined I think we need to make sure all the required parameters are definied.

Reply:
right, its probably a good idea to make the Endpoint required, the `IPAddressPool` is already. Initially I was thinking we make the `Endpoint` optional to be able to expose the api via a service to any network, but only register the endpoint on the net where `Endpoint` is specified. would that use case make sense? I am not sure right now if it is needed

Reply:
Oh I see your point: I'm not sure either atm, it would be nice having the ability to register the endpoint on the net where the `Endpoint` is specified, but we can take this as a follow up in case it will be a pattern followed by the other operators.